### PR TITLE
Dev narrative

### PR DESCRIPTION
**Pull request by TimAlanWhite**

### CODE REVIEW

File: world/quests/test/task-demo.lisp [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+| ! eval (complete 'fi-check)
+| ! eval (activate 'fi-outcome)")
+(quest:interaction :name home-check :interactable home :dialogue "
+~ player
+| (They made a home here. Who was Harris to take that away?)
+| ! eval (complete 'fi-check)
+")
+(quest:interaction :name home-tent-check :interactable home-tent :dialogue "
+~ player
+| (Androids do use bedrolls. At least, this one did.)
```
Comment (Shinmera) [inline world/quests/test/task-demo.lisp:158]: `Huh, so some androids do sleep after all. And given everything else, I don't think this one did it to blend in, either.`
`Perhaps it brought them comfort?`

File: world/quests/test/task-demo.lisp [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+| The android then killed Harris - not just killed him, but dismembered him.
+| They spent their final seconds crawling back into their home, where they offlined.
+~ fi
+| -- You think Harris tricked us into coming here? To seize you too?
+~ player
+| Perhaps.
+~ fi
+| I see you have the memory replay.
+~ player
+| *Holds the small metal chip aloft*
```
Comment (Shinmera) [inline world/quests/test/task-demo.lisp:182]: `Got it right here.`

File: world/quests/test/task-demo.lisp [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+~ fi
+| Let's go home. We'll see what it can tell us about the past.
```
Comment (Shinmera) [inline world/quests/test/task-demo.lisp:184]: `Alright, let's go home. We'll see what it can tell us about the past.`

File: world/quests/test/task-demo.lisp [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+| By the lay of the carnage in there, I'd say Harris and company ambushed them.
+| They fought, and the cave destabilised and collapsed. The humans were killed in the cave in - all but one.
+| The android then killed Harris - not just killed him, but dismembered him.
+| They spent their final seconds crawling back into their home, where they offlined.
+~ fi
+| -- You think Harris tricked us into coming here? To seize you too?
+~ player
+| Perhaps.
+~ fi
+| I see you have the memory replay.
```
Comment (Shinmera) [inline world/quests/test/task-demo.lisp:180]: `Did you get the Genera core?`

**Review by Shinmera — CHANGES_REQUESTED** [review]
Please see my detailed comments in the change view. 

Before we can merge, you should also rebase your branch onto master to get the new commits that add some of the features I talk about in the comments. You'll have to do a force-push after the rebase, so make sure you don't have any unpulled commits before you rebase.

File: world/quests/test/task-demo.lisp [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+| Tracking the android was my idea. Talking to Harris was my idea. Clinging to your back while you scale a vertical cliff face - (:shocked)definitely not my idea.
+~ player
+| Well, we're here now. Can I proceed?
+~ fi
+| You know you don't need to ask me that, of all people? But yes, you are permitted.
+| Though I'm staying right here. Let me know what you find. And be careful.")
+(quest:interaction :name pot-check :interactable pot :dialogue "
+~ player
+| (The cooking pot infers human habitation.
+| (And this is a great spot for a lookout...)
```
[the remark this answers is not in the document]
Comment (TimAlanWhite) [reply world/quests/test/task-demo.lisp:22]: sounds good. I think I agree with the parentheticals being too distracting, especially since thoughts are going to be occurring in the field potentially more than speech
The suggested syntax would work for me +1

File: world/quests/test/task-demo.lisp [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+| ! eval (activate 'fi-tent)
+| ! eval (complete 'fi-check)
+")
+; | ! eval (let (fi-tent-talk true)) - works to set
```
[the remark this answers is not in the document]
Comment (TimAlanWhite) [reply world/quests/test/task-demo.lisp:26]: ah, as suspected, LISP fail :) That's cool
We could just leave the guarding for now, and allow dialogues to repeat as default, even though for some it doesn't make sense. Somewhat of an edge case for most players I suspect anyway
Or if time I could refactor to use the quests and tasks properly - though might be better waiting till the vertical slice for that kind of thing

File: world/quests/test/task-demo.lisp [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+(quest:interaction :name tent-check :interactable tent :dialogue "
+~ player
+| (There's no one inside, but the bedroll suggests it was recently occupied.)
+| (Does an android require a bedroll? I recharge standing up.)
```
[the remark this answers is not in the document]
Comment (TimAlanWhite) [reply world/quests/test/task-demo.lisp:30]: I like the suggested edit 👍 
"People/persons" sounds good

Unsure at the literal start of the game whether the Stranger would distance herself from people - would need more thought in consideration of the plot and character development. However, for this demo I think it could be an interesting hook right from the start to set the scene of the android being in service to the people to a degree.

This quest might not be in the final game, but if so it feels like something mid-game, which I don't think is a bad thing to drop the demo player into the middle of - makes it feel like stuff has already happened and they're straight into a cool mission (as long as there is enough orientation/tutorialisation still)

File: world/quests/test/task-demo.lisp [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+(quest:interaction :name fi-sign-main :interactable fi :dialogue "
+~ player
+| That sign over there has the same symbol on it as my uniform. I think it has something to do with androids.
+~ fi
+| Then we're on the right track.
+~ player
+| It also resembles an arrow, don't you think? Pointing up over the mountain?
+~ fi
+| I know you want to be a detective, but now I think you're seeing things.
+| Remind me to check your pattern recognition software when we get back.
```
[the remark this answers is not in the document]
Comment (TimAlanWhite) [reply world/quests/test/task-demo.lisp:60]: Cool, yeah I realise I started playing fast and loose with the lore at this point as well :)

File: world/quests/test/task-demo.lisp [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+| [has-more-dialogue You got more? | You'd better get going.]")
+(quest:interaction :name sign-check :interactable sign :dialogue "
+~ player
+| (The words were scratched into the wood with a knife. I can barely read it. Upscaling resolution and enabling zoom...)
```
[the remark this answers is not in the document]
Comment (TimAlanWhite) [reply world/quests/test/task-demo.lisp:64]: This is a direct quote I believe? In which I'm wary of such a direct reference - it's fairly obscure and is small, so plagiarism probably isn't an issue. And I think pop culture references can serve as inter-textual, postmodern effects; but in general I don't like direct lifts like this, as it doesn't feel creative.

I think you're right in that something shorter here would be punchier - I don't like my original, looking at it with fresh eyes. We could keep the enhance aspect from Blade Runner, but simply have something like:

> Enhancing resolution...

The only point really is to reinforce she's a machine while in the field from time to time

File: world/quests/test/task-demo.lisp [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+| //READ ME - NO LITTER U FUCKS - BOTS PATROL HERE//
+| (Short and not so sweet - that's Harris.)
```
[the remark this answers is not in the document]
Comment (TimAlanWhite) [reply world/quests/test/task-demo.lisp:66]: The 3 way check sounds interesting, I'm sure there could end up being use-cases for it.
Harris' name is mentioned in the initial dialogue with Fi, and a couple of other times before the sign, depending on what the player interacts with.
Agree, without that, this does sound weird

File: world/quests/test/task-demo.lisp [hunk 11]
```diff
@@ -0,0 +1,185 @@
...
+~ fi
+| That's not unusual in these parts. But given that everything else seems wrong about this place...
+~ player
+| I hope the android isn't buried under there.
+~ fi
+| Androids are tougher than they look.
+~ player
+| I don't look tough?
+~ fi
+| *Smiles* I didn't say that.
```
[the remark this answers is not in the document]
Comment (TimAlanWhite) [reply world/quests/test/task-demo.lisp:100]: Yeah I figured we'd probably want to handle something like this with a bespoke chat head emote, but it felt like a nice dramatic action to call out at the time

Could there also be times where we want to portray actions between characters that we can't animate? E.g. at the end I have the Stranger holding a chip aloft. It would allow us to visualise nice character details, while the in-game models are frozen, without incurring art budget. I suppose bringing in a little bit of prose narration for the player's imagination.

Asterisking them looks kind of ugly too I think, though I've seen other games do this in their subtitles, and I think players understand what it means

File: world/quests/test/task-demo.lisp [hunk 12]
```diff
@@ -0,0 +1,185 @@
...
+| I think it's fair to say that Harris hasn't been completely honest with us.
+~ player
+| You think it's a trap?
+~ fi
+| There's one way to find out.
+~ player
+| Yes... Lucky me.
+~ fi
+| You can handle yourself.
+| [has-more-dialogue Next? | Sayonara.]")
```
[the remark this answers is not in the document]
Comment (TimAlanWhite) [reply world/quests/test/task-demo.lisp:121]: heh, nice, yeah I do like a bit of language code-switching from time to time, to show more accent and dialect. Stylistically for consistency I think it would still be written in English here

File: world/quests/test/task-demo.lisp (hunk 7 above)
[the remark this answers is not in the document]
Comment (Shinmera) [reply world/quests/test/task-demo.lisp:30]: Sounds good.

File: world/quests/test/task-demo.lisp (hunk 9 above)
[the remark this answers is not in the document]
Comment (Shinmera) [reply world/quests/test/task-demo.lisp:64]: Fair enough.

File: world/quests/test/task-demo.lisp (hunk 11 above)
[the remark this answers is not in the document]
Comment (Shinmera) [reply world/quests/test/task-demo.lisp:100]: I'll be frank, when I saw the emote I cringed pretty hard. I think it just comes across as amateurish, especially when we have the emote profile system already in place.

I understand that players get what it means, it's a common technique in text chats and so forth, especially in times before emojis. Still, I think adding a profile animation when necessary, just leaving it out, or expressing it in dialogue would ultimately be better.

It's especially jarring when it's a very seldom used thing, since you aren't prepared for dialogue suddenly being used for actions as well.

File: world/quests/test/task-demo.lisp [hunk 13]
```diff
@@ -0,0 +1,185 @@
...
+| (The scratches in the rocks suggest combat occurred - swords, gunfire, and finger nails...)
+| ! eval (complete 'fi-check)
+")
+(quest:interaction :name android-check :interactable android :dialogue "
+~ player
+| (It's the android... It's been destroyed. Not even Catherine could fix this kind of damage.)
+| Hello, sestra... What happened to you?
+| (More human remains. It appears you didn't go down without a fight.)
+| (The memory replay seems to be intact.)
+| Excuse me, sestra... I just need to retrieve this... --- Thank you.
```
[the remark this answers is not in the document]
Comment (TimAlanWhite) [reply world/quests/test/task-demo.lisp:147]: cool. I'm reminded as well I went with russian dialect word for sister here. Obviously the russian style of voice for the stranger is something I pitched, and I kind of had in my head when writing the lines in this quest, to subtly inform word choice, with making it weird with lots of stranger phonetic spellings.
Still not sure how I feel about it myself, and something definitely to discuss later when we move onto the vertical slice

File: world/quests/test/task-demo.lisp (hunk 1 above)
Comment (TimAlanWhite) [reply world/quests/test/task-demo.lisp:158]: yeah, I thought this was a nice character detail and callback to the beginning, so a bit more rumination makes sense here 👍 
Dunno if you've seen my level design "skills", but I just copied the tent and put it in the cave :)

File: world/quests/test/task-demo.lisp (hunk 4 above)
Comment (TimAlanWhite) [reply world/quests/test/task-demo.lisp:180]: aha, lore!

File: world/quests/test/task-demo.lisp (hunk 13 above)
[the remark this answers is not in the document]
Comment (Shinmera) [reply world/quests/test/task-demo.lisp:147]: When reading this I first thought of something else that could be interesting to explore instead, namely that the androids developed some private vocabulary that they use for each other. We could use words from other languages for that, or other adjacent, uncommon terms from English.

As you said though, at this point I'm not so sure either what the best direction is here.

File: world/quests/test/task-demo.lisp (hunk 1 above)
Comment (Shinmera) [reply world/quests/test/task-demo.lisp:158]: It looked fine, if a bit stark :)

File: world/quests/test/task-demo.lisp (hunk 2 above)
Comment (TimAlanWhite) [reply world/quests/test/task-demo.lisp:182]: I do like the freedom the action descriptions would give us to show little visual actions from time to time, without having to rely on speech to convey it instead (and without having to animate it necessarily - though bonus if the art budget can do that :) ). Though is perhaps a wider discussion, including on how a narrative delivery method like that should be implemented in a system so far designed for speech

Comment (Shinmera) [reply world/quests/test/task-demo.lisp:182]: I think the Ace Attorney games are a perfect example on how to handle this situation. They have very, very few animations for every character, and I don't believe they ever rely on asterisks to express actions, instead using UI elements, or context clues in a very effective way.

Here's a good video on their animations in general: https://www.youtube.com/watch?v=zV8sjRTiWC8

File: world/quests/test/task-demo.lisp [hunk 14]
```diff
@@ -0,0 +1,185 @@
...
+| (Perhaps it's a warning...) 
+| ! eval (activate 'fi-sign-main)
+| ! eval (complete 'fi-check)")
+(quest:interaction :name fi-sign-main :interactable fi :dialogue "
+~ player
+| That sign over there has the same symbol on it as my uniform. I think it has something to do with androids.
+~ fi
+| Then we're on the right track.
+~ player
+| It also resembles an arrow, don't you think? Pointing up over the mountain?
```
[the remark this answers is not in the document]
Comment (TimAlanWhite) [reply world/quests/test/task-demo.lisp:57]: Slightly tweaked further:

> Although, it's not painted in the correct color. Perhaps it has another meaning... A directional arrow?
